### ISSUE TITLE
Replicator in fieldset causes error when trying to link or edit

### ISSUE THREAD

**Issue by juresrpcic** [statamic/cms#3136]
## Bug Description
This issue is related to #2942 and #2924 - I've just investigated a bit more to see where the problem is.

## How to Reproduce

1. Create replicator inside fieldset (works if replicator is added directly, not from fieldset)
2. Add at least one field to replicator (works as long as replicator inside fieldset is empty)
3. Try to add replicator to a blueprint - the screen is stuck on "Loading"

## Extra Detail

- the backend call to `/cp/fields/edit` gives an error (500)

![image](https://user-images.githubusercontent.com/2803433/105203064-f4ebdf00-5b42-11eb-9952-2b1876828a2f.png)

- the error is located in  `/cms/fields/FieldTransformer.php` - line 61 - `toVue()` method

```public static function toVue($field): array
    {
        if (isset($field['import'])) {
            return static::importFieldToVue($field);
        }

        return is_string($field['field'])
            ? static::referenceFieldToVue($field)
            : static::inlineFieldToVue($field);
    }
```

- examining the payload the issue seems to be that Vue is sending this kind of array directly when the error is encountered (no $field found - as error says):

```array:18 [
  "options" => array:2 [
    "d1" => "D1"
    "d2" => "D2"
  ]
  "multiple" => false
  "max_items" => 1
  "clearable" => false
  "searchable" => false
  "taggable" => false
  "push_tags" => false
  "cast_booleans" => false
  "display" => "Grid one design"
  "type" => "select"
  "icon" => "select"
  "listable" => "hidden"
  "placeholder" => null
  "component" => "select"
  "handle" => "grid_one_design"
  "prefix" => null
  "instructions" => null
  "required" => false
]
```

while when editing the same type of replicator added directly to blueprint (not through fieldset) sends this type of array:

```array:2 [
  "handle" => "choose_design"
  "field" => array:12 [
    "options" => array:2 [
      1 => "1"
      2 => "2"
    ]
    "multiple" => false
    "max_items" => 1
    "clearable" => false
    "searchable" => false
    "taggable" => false
    "push_tags" => false
    "cast_booleans" => false
    "display" => "Choose design"
    "type" => "select"
    "icon" => "select"
    "listable" => "hidden"
  ]
```

I'm dumping these arrays from `toVue()` so it's finding the first field, just fyi:
![image](https://user-images.githubusercontent.com/2803433/105204292-4fd20600-5b44-11eb-90c2-9d43a08546ec.png)

I'm guessing the wrong function is called - it probably shouldn't be `toVue()` but something else - but not sure how this is wired. It's a tough issue though as it's stopping me from making my page builder (I'm currently resorting to building little pieces elsewhere and then directly editing YAMLs). 

There are some other places where the same issue happens, but it appears to me that if this is solved here it's going to resolve those issues too so I'm not listing them (just similar nesting stuff - it happens also when editing something nested similarly inside Fieldset, not working on Blueprints at all).

## Environment

**Statamic version:** 3.0.39
**PHP version:** 7.4.13

**Install method (choose one):**

- Existing Laravel app + eloquent driver (completely fresh install for testing this)

**Comment by juresrpcic** [comment]
@damcclean Could you also tag this with `bug` - it's really breaking the ability to use fieldsets for even relatively simple fields, a very core feature.

**Comment by damcclean** [comment]
> @damcclean Could you also tag this with `bug` - it's really breaking the ability to use fieldsets for even relatively simple fields, a very core feature.

I'll need to test it first before tagging it with `bug`. That label is only for confirmed bugs.

**Comment by juresrpcic** [comment]
Aha sure. Fyi for testing - all steps above still create the bug - but in the meantime I discovered I can do any type of deep nesting when attaching all fields directly to the collection, bypassing Fieldsets entirely.

It just means that any field sharing doesn't work (ie. I need the Page Builder Replicator field on different collections) so I need to manually copy the whole field definition to each collection that needs it on every change.

And additionally I discovered that removing a field inside a fieldset breaks editing of entries with it completely - but if not using fieldsets, it just leaves an empty block "tab" there, still allowing to edit all other fields. It's probably related.
